--- a/src/lib/formats/flopimg.h
+++ b/src/lib/formats/flopimg.h
@@ -16,6 +16,7 @@
 #include "utilfwd.h"
 
 #include <memory>
+#include <string_view>
 #include <vector>
 
 #include <cassert>
--- a/src/lib/formats/rpk.h
+++ b/src/lib/formats/rpk.h
@@ -19,6 +19,7 @@
 #include <cassert>
 #include <list>
 #include <optional>
+#include <vector>
 
 
 /***************************************************************************
--- a/src/osd/modules/sound/coreaudio_sound.cpp
+++ b/src/osd/modules/sound/coreaudio_sound.cpp
@@ -20,6 +20,8 @@
 #include <CoreFoundation/CoreFoundation.h>
 #include <CoreServices/CoreServices.h>
 
+#include <atomic>
+#include <mutex>
 #include <map>
 #include <memory>
 #include <new>
--- a/3rdparty/ymfm/src/ymfm_opq.cpp
+++ b/3rdparty/ymfm/src/ymfm_opq.cpp
@@ -27,6 +27,8 @@
 // CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
 // OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
 // OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+#include <cstdlib>
 
 #include "ymfm_opq.h"
 #include "ymfm_fm.ipp"
