--- a/Source/Core/VideoCommon/TextureCacheBase.cpp
+++ b/Source/Core/VideoCommon/TextureCacheBase.cpp
@@ -547,7 +547,8 @@
         return std::nullopt;
       }
 
-      tex->texture->Load(level, level_width, level_height, level_width, &texture_data[start], size);
+      tex->texture->Load(level, level_width, level_height, level_width, &texture_data[start], size,
+                         layer);
       start += size;
     }
   }
